### PR TITLE
WIREUP: Restrict creation more BW lanes than previously

### PR DESCRIPTION
**Pull request by ivankochin**
## What
Restrict creation more BW lanes than during first lanes initialization.

## Why ?
To prevent endpoint reconfiguration error in case when some BW lanes were dropped due to low BW during first initialization, but there are available resources with higher BW that still can be chosen instead of dropped lanes during second initialization.

This is the short-term solution. The long-term solution is to support endpoint reconfiguration.

### CODE REVIEW

File: src/ucp/wireup/select.c [hunk 1]
```diff
@@ -1508,6 +1508,30 @@ ucp_wireup_add_fast_lanes(ucp_worker_h worker,
     return num_lanes;
 }
 
+/* Restrict choosing more lanes that were chosen previously to prevent
+ * endpoint reconfiguration error in case of dropping some lanes due to low BW.
+ */
+static void
+ucp_wireup_bw_lanes_check_prev(const ucp_wireup_select_params_t *select_params,
```
Comment (yosefe) [inline src/ucp/wireup/select.c:1515]: `static unsigned ucp_wireup_get_num_lanes(const ucp_ep_config_key_t *ep_cfg_key, ucp_lane_type lane_type)`

Comment (ivankochin) [reply src/ucp/wireup/select.c:1515]: As I understand, it is not guaranteed that EP would have assigned key in that point, so we cannot just use `ucp_ep_config(ep)` to pass it. Do you suggest to move `(ep->cfg_index != UCP_WORKER_CFG_INDEX_NULL)` outside of this function or maybe pass key to `ucp_wireup_search_lanes` as argument?

Comment (yosefe) [reply src/ucp/wireup/select.c:1515]: yes, move it outside the function

Comment (ivankochin) [reply src/ucp/wireup/select.c]: As it was discussed, changed  the return type to unsigned + passed EP as input parameter.

**Review by yosefe — APPROVED** [review]

**Pull request merged**